code/ship/ship.cpp

```diff
@@ -5634,29 +5640,31 @@ void physics_ship_init(object *objp)
 		nprintf(("Physics", "pi->mass==0.0f. setting to %f\n",amass));
 		Warning(LOCATION, "%s (%s) has no mass! setting to %f", sinfo->name, sinfo->pof_file, amass);
 		pm->mass=amass;
-		pi->mass=amass*sinfo->density;
+		pi->mass=amass * sinfo->density;
 	}
+	else 
+		pi->mass = pm->mass * sinfo->density;
 
-	// ack!
-	// if pm's MOI is invalid, compensate
-	if ( IS_VEC_NULL(&pm->moment_of_inertia.vec.rvec)
+	// it was print-worthy back in modelread.cpp, but now that its being used for an actual ship the user should be warned.
```

Comment:
modelread.cpp is a big file; you should say which function that print appears in